src/index.js

```diff
@@ -66,18 +66,26 @@ class BabelEsmPlugin {
         }
       }
 
-      Object.keys(compiler.options.entry).forEach(entry => {
-        const entryFiles = compiler.options.entry[entry];
-        if (Array.isArray(entryFiles)) {
-          new MultiEntryPlugin(compiler.context, entryFiles, entry).apply(
-            childCompiler,
-          );
-        } else {
-          new SingleEntryPlugin(compiler.context, entryFiles, entry).apply(
-            childCompiler,
-          );
-        }
-      });
+      if (typeof compiler.options.entry === 'string') {
+        new SingleEntryPlugin(
+          compiler.context,
+          compiler.options.entry,
+          'index',
+        ).apply(childCompiler);
+      } else {
+        Object.keys(compiler.options.entry).forEach(entry => {
+          const entryFiles = compiler.options.entry[entry];
+          if (Array.isArray(entryFiles)) {
+            new MultiEntryPlugin(compiler.context, entryFiles, entry).apply(
+              childCompiler,
+            );
+          } else {
+            new SingleEntryPlugin(compiler.context, entryFiles, entry).apply(
+              childCompiler,
+            );
+          }
+        });
+      }
```

Comment:
how about merging the branches here?

i.e.

`if (typeof compiler.options.entry === 'string') compiler.options.entry = { index: compiler.options.entry };`

possibly extracting the var first

Reply:
done

Reply:
as long as overriding it doesn't affect something

Reply:
i guess the tests cover that, right?

Reply:
@AlexMost WDYT?

Reply:
looks good, thanks!